Scripts/RemediationScripts/Remediate-SetServiceBusMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,813 @@
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+        }
+        else {
+            if(-not $AutoRemediation){
+                Write-Host "'Force' flag is provided. Secure TLS version will be enabled on Azure Service Bus Namespace(s) in the Subscription without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+
+        # List for storing remediated Azure Service Bus Namespace(s)
+        $serviceBusNamespacesRemediated = @()
+
+        # List for storing skipped Azure Service Bus Namespace(s)
+        $serviceBusNamespacesSkipped = @()
+
+        Write-Host "Setting secure TLS version on Azure Service Bus Namespace(s)..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Loop through the list of Azure Service Bus Namespace(s) which needs to be remediated.
+        $actionableServiceBusNamespacesDetails  | ForEach-Object {
+            try {
+                $updatedTLSVersion = (Set-AzServiceBusNamespace -ResourceGroupName $_.ResourceGroupName -NamespaceName $_.ResourceName -MinimumTlsVersion $([Constants]::MinRequiredTLSVersionValue)).MinimumTlsVersion 
+
+                if (CheckIfCurrentTLSVersionIsSecure($updatedTLSVersion)) {
+                    $_.MinimumTLSVersion = $updatedTLSVersion
+                    $serviceBusNamespacesRemediated += $_
+                    $logResource = @{}	
+                    $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                    $logResource.Add("ResourceName", ($_.ResourceName))	
+                    $logRemediatedResources += $logResource	
+                }
+                else {
+                    $serviceBusNamespacesSkipped += $_
+                    $logResource = @{}	
+                    $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                    $logResource.Add("ResourceName", ($_.ResourceName))
+                    $logResource.Add("Reason", "Error setting server parameter MinimumTLSVersion: [$($_)]")            
+                    $logSkippedResources += $logResource	
+
+                }                
+            }
+            catch {
+                $serviceBusNamespacesSkipped += $_
+                $logResource = @{}	
+                $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+                $logResource.Add("ResourceName", ($_.ResourceName))	
+                $logResource.Add("Reason", "Encountered error while setting server parameter MinimumTLSVersion")    	
+                $logSkippedResources += $logResource	
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)	
```

Comment:
since you are not printing anything like the script is going to set min TLS version for this resource.
Does printing "Skipping this resource..." make sense?

Reply:
Good point, this might cause some confusion as well, removing it.